src/parse.ts

```diff
@@ -205,8 +235,8 @@ export const parseFormulaIdentifier = (variableName: string, originalObject: str
   if (isCustomSetting(variableName)) {
     return parseCustomSetting(variableName)
   }
-  if (isRelationshipField(variableName)) {
-    return parseRelationship(variableName, originalObject)
+  if (isNameWithComponents(variableName)) {
+    return [parseRelationship(variableName, originalObject), parseInstance(variableName)].flat()
```

Comment:
Why not encapsulate parseInstance in parseRelationship?

Reply:
I feel like they handle different cases? Also, `parseRelationship` seems complicated enough already?